go/mobile/identity.go

```diff
@@ -0,0 +1,208 @@
+	idenPubOnChain, keyStore, storage, err := loadComponents(storePath, web3Url)
+	if err != nil {
+		return id, err
+	}
+	resourcesAreClosed := false
+	defer func() {
+		if !resourcesAreClosed {
+			keyStore.Close()
+			storage.Close()
+		}
+	}()
+	// Create babyjub keys
+	kOpComp, err := keyStore.NewKey([]byte(pass))
+	if err != nil {
+		return id, err
+	}
+	if err = keyStore.UnlockKey(kOpComp, []byte(pass)); err != nil {
+		return id, err
+	}
+	// Store kOpComp
+	tx, err := storage.NewTx()
+	if err != nil {
+		return id, err
+	}
+	if err := db.StoreJSON(tx, []byte("kOpComp"), kOpComp); err != nil {
+		return id, err
+	}
+	// Init claim DB
+	tx.Put([]byte("receivedCredentialsCounter"), []byte("0"))
+	if err := tx.Commit(); err != nil {
+		return id, err
+	}
+	// Parse extra genesis claims
+	// TODO: Call toClaimers once it's implemented
+	// _extraGenesisClaims, err := extraGenesisClaims.toClaimers()
+	if err != nil {
+		return id, err
+	}
+	// Create new Identity (holder)
+	_, err = holder.New(
+		holder.ConfigDefault,
+		kOpComp,
+		nil,
+		storage,
+		keyStore,
+		idenPubOnChain,
+		nil,
+		readerhttp.NewIdenPubOffChainHttp(),
+	)
+	if err != nil {
+		return id, err
+	}
+	// Verify that the Identity can be loaded successfully
+	keyStore.Close()
+	storage.Close()
+	resourcesAreClosed = true
+	return NewIdentityLoad(storePath, pass, web3Url, checkTicketsPeriodMilis, e)
+}
+
+// NewIdentityLoad loads an already created identity
+// this funciton is mapped as a constructor in Java
+func NewIdentityLoad(storePath, pass, web3Url string, checkTicketsPeriodMilis int, e Event) (*Identity, error) {
+	// TODO: figure out how to diferentiate the two constructors from Java: https://github.com/iden3/iden3-mobile/issues/17#issuecomment-587374644
+	id := &Identity{}
+	idenPubOnChain, keyStore, storage, err := loadComponents(storePath, web3Url)
+	if err != nil {
+		return id, err
+	}
+	defer keyStore.Close()
+	// Unlock key store
+	kOpComp := &babyjub.PublicKeyComp{}
+	if err := db.LoadJSON(storage, []byte("kOpComp"), kOpComp); err != nil {
+		return id, err
+	}
+	if err := keyStore.UnlockKey(kOpComp, []byte(pass)); err != nil {
+		return nil, fmt.Errorf("Error unlocking babyjub key from keystore: %w", err)
+	}
+	// Load existing Identity (holder)
+	holdr, err := holder.Load(storage, keyStore, idenPubOnChain, nil, readerhttp.NewIdenPubOffChainHttp())
+	if err != nil {
+		return id, err
+	}
+	// Init Identity
+	id.id = holdr
+	id.Tickets = &TicketsMap{
+		m: make(map[string]*Ticket),
+	}
+	id.eventSender = e
+	id.storage = storage
+	id.loadTickets()
+	go id.checkPendingTickets(time.Duration(checkTicketsPeriodMilis * 1000000))
+	return id, nil
+}
+
+// TODO: update marshal functions (s *Struct) ==> (s Struct)
+
+// Stop close all the open resources of the Identity
+func (i *Identity) Stop() {
+	log.Info("Stopping identity: ", i.id.ID())
+	defer i.storage.Close()
+	// Send "singnal" to stop the pending tickets loop
+	i.Tickets.shouldStop = true
```

Comment:
better to use channels instead of reading and writing to variables concurrently.  Example (see `Start` and `StopAndJoin`): https://github.com/iden3/go-iden3-servers/blob/master/loaders/loaders.go#L249

Reply:
Last update already uses channel